gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/MigrateApiUseCaseTest.java

```diff
@@ -256,6 +268,182 @@ void should_return_fail_when_api_dont_use_v4_emulation_engine() {
             .containsExactly(new MigrationResult.Issue("Cannot migrate an API not using V4 emulation", MigrationResult.State.IMPOSSIBLE));
     }
 
+    @Test
+    void should_migrate_when_api_has_markdown_page() {
+        // Given
+        var api = ApiFixtures.aProxyApiV2().toBuilder().id(API_ID).build();
+        api.getApiDefinition().setExecutionMode(ExecutionMode.V4_EMULATION_ENGINE);
+        PageEntity publishedPage = new PageEntity();
+        publishedPage.setPublished(true);
+        PageEntity markdownTemplatePage = new PageEntity();
+        markdownTemplatePage.setPublished(true);
+        markdownTemplatePage.setType(PageType.MARKDOWN_TEMPLATE.name());
+        doReturn(Arrays.asList(publishedPage, markdownTemplatePage)).when(pageService).findByApi(any(), any());
```

Comment:
For maintenability it’s preferable to write it like:


```suggestion
        when(pageService.findByApi(any(), any())).thenReturn(Arrays.asList(publishedPage, markdownTemplatePage));
```

(same for others tests cases)

Reply:
fixed